src/AssetsTransferApi.spec.ts

```diff
@@ -157,13 +167,179 @@ describe('AssetTransferAPI', () => {
 			});
 		});
 	});
+
+	describe('fetchCallType', () => {
+		describe('RelayToSystem', () => {
+			it('Should correctly return Teleport', () => {
+				const assetCallType = relayAssetsApi['fetchCallType'](
+					'0',
+					'1000',
+					['ksm'],
+					Direction.RelayToSystem,
+					AssetType.Native,
+					false,
+					relayAssetsApi.registry
+				);
+
+				expect(assetCallType).toEqual('Teleport');
+			});
+		});
+		describe('RelayToPara', () => {
+			it('Should correctly return Reserve', () => {
+				const assetCallType = relayAssetsApi['fetchCallType'](
+					'0',
+					'2023',
+					['ksm'],
+					Direction.RelayToPara,
+					AssetType.Native,
+					false,
+					relayAssetsApi.registry
+				);
+
+				expect(assetCallType).toEqual('Reserve');
+			});
+		});
+		describe('SystemToRelay', () => {
+			it('Should correctly return Teleport', () => {
+				const assetCallType = systemAssetsApi['fetchCallType'](
+					'1000',
+					'0',
+					['ksm'],
+					Direction.SystemToRelay,
+					AssetType.Native,
+					false,
+					systemAssetsApi.registry
+				);
+
+				expect(assetCallType).toEqual('Teleport');
+			});
+		});
+		describe('SystemToSystem', () => {
+			it('Should correctly return Teleport when sending a native asset', () => {
+				const assetCallType = systemAssetsApi['fetchCallType'](
+					'1000',
+					'1001',
+					['ksm'],
+					Direction.SystemToSystem,
+					AssetType.Native,
+					false,
+					systemAssetsApi.registry
+				);
+
+				expect(assetCallType).toEqual('Teleport');
+			});
+			it('Should correctly return Reserve when sending a foreign asset', () => {
```

Comment:
I am not totally sure about this assumption here. Also, at the moment there such scenario (sending foreign assets between system paras) is not possible in the ecosystem. I would remove this spec